lib/cloudkeeper/aws/image_downloader.rb

```diff
@@ -0,0 +1,42 @@
+require 'net/https'
+
+module Cloudkeeper
+  module Aws
+    # Class used for downloading images for cloudkeeper
+    class ImageDownloader
+      # Downloads file from uri by segments
+      #
+      # @param image_uri [String] uri of the image to download
+      # @raise [Cloudkeeper::Aws:Errors::ImageDownloadError] if download failed
+      def self.download(image_uri, limit = 10, &block)
+        raise Cloudkeeper::Aws::Errors::ImageDownloadError, 'Too many redirects' \
+          if limit.zero?
+
+        uri = URI.parse(image_uri)
```

Comment:
I'm not sure from where will you get that `image_uri` but you should probably catch `URI::InvalidURIError` which can be risen by `URI.parse` method and handle it somehow.